src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -41,8 +41,10 @@ internal class DefaultCircuitFactory : CircuitFactory
             var encoder = scope.ServiceProvider.GetRequiredService<HtmlEncoder>();
             var jsRuntime = (RemoteJSRuntime)scope.ServiceProvider.GetRequiredService<IJSRuntime>();
             var componentContext = (RemoteComponentContext)scope.ServiceProvider.GetRequiredService<IComponentContext>();
+            var authenticationStateProvider = (FixedAuthenticationStateProvider)scope.ServiceProvider.GetRequiredService<IAuthenticationStateProvider>();
             jsRuntime.Initialize(client);
             componentContext.Initialize(client);
+            authenticationStateProvider.Initialize(httpContext.User);
```

Comment:
Side note: We keep running into these type of patterns where we need to initialize stuff at some point in the lifecycle. I'm wondering if we are missing an abstraction here (that can potentially also be useful for customers).

It feels to me that we are missing some sort of platform "initializer" with maybe individual initializers for different platforms. Something that we and people can implement and that will get resolved from DI and called in at the right time to initialize platform dependent stuff like Auth, the UriHelper or any primitive that is context dependent like browser abstractions.

Reply:
If the need for some new abstraction arises I'm sure we can.

I hope that for customers, our existing DI APIs cover most requirements. For example, to have an `IAuthorizationService` that obtains data from some other DI service (e.g., an `IHttpContextAccesstor`), the existing DI APIs allow the construction of a scoped service in terms of another scoped/singleton service.

Reply:
The Fowler will come to haunt you in your dreams if you use IHttpContextAccessor. 
But I don't think DI is enough for this because different platforms have different requeriments, like interactive circuits vs statically rendered components, and DI doesn't enable you to have both registered at the same time. (And also DI only solves the where to get things from, but not the when is the right time to initialize things).

I'm just saying that things like these keep piling up.

Reply:
Personally I think the current state is the **best possible** solution. 

It's not overdesigned. It's very clear to see what happens and in what order. If the ordering changes, or the dependencies change, then that will be reflected in a clear diff.

I don't think we should turn this into a filter pipeline, or property bag, or anything like that. To me simple is good.

Reply:
Marking resolved because I think we're concluded that we're not changing this in this PR.